torch/csrc/autograd/profiler.h

```diff
@@ -273,6 +328,33 @@ struct TORCH_API RecordProfile {
   void processEvents(const std::vector<Event*>& events);
 };
 
+// A guard that enables the profiler, taking in an optional callback to process
+// the results
+// Usage:
+// {
+//   TLSProfilerGuard g([](thread_event_lists profilerResults) {
+//     // process profilerResuls
+//   });
+//   Code to profile
+// }
+struct TORCH_API TLSProfilerGuard {
```

Comment:
I guess we could impl. RecordProfile though this, but we can do it in a later PR